mainApp/src/main/java/logisticsmarshall/tqs/ua/controllers/LogisticsAPIController.java

```diff
@@ -158,6 +160,28 @@
         return ResponseEntity.ok(rep);
     }
 
+    @PostMapping(path="/reputation",consumes = "application/json")
+    public ResponseEntity<Reputation> postRating(@RequestParam(name = "rating") int rating,
+                                                   @RequestParam(name = "delivery_id") int delivery_id,
+                                                   @RequestParam(name = "driver_id") int driver_id,
+                                                   @RequestParam(name = "apiKey") String apikey,
+                                                   @RequestParam(name = "description") String description) {
+        Company companyFromapiKey = deliveryService.getApiKeyHolderCompany(apikey);
+        if (companyFromapiKey == null) return ResponseEntity.status(403).build();
+        Delivery delRequested = deliveryService.getDeliveryById(delivery_id);
+        Driver driverAssigned = driverService.getDriverById(driver_id);
+
+        Reputation rep = new Reputation();
+        rep.setDelivery(delRequested);
+        rep.setRating(delivery_id);
+        rep.setDriver(driverAssigned);
+        rep.setDescription(description);
+        rep.setRating(rating);
+
```

Comment:
you probably need to set delivery.reputation and save it too or it'll fail when overwriting a review

Reply:
sure